build/org.eclipse.pde.build/src/org/eclipse/pde/internal/build/site/BuildTimeFeature.java

```diff
@@ -156,31 +153,28 @@ public BuildTimeSiteContentProvider getFeatureContentProvider() {
 		return contentProvider;
 	}
 
-	public URL getURL() {
-		return url;
+	public Path getPath() {
+		return path;
 	}
 
-	public void setURL(URL url) {
-		this.url = url;
+	public void setPath(Path path) {
+		this.path = path;
 	}
 
 	/**
 	 * @return the local filesystem location of the directory containing the feature.xml file. 
 	 */
 	public String getRootLocation() {
 		if (rootLocation == null) {
-			URL location = getURL();
-			if (location == null)
+			Path location = getPath();
+			if (location == null) {
 				return null;
-			try {
-				URI locationURI = URIUtil.toURI(location);
-				rootLocation = URIUtil.toFile(locationURI).getAbsolutePath();
-			} catch (URISyntaxException e) {
-				rootLocation = location.getPath();
 			}
-			int i = rootLocation.lastIndexOf(Constants.FEATURE_FILENAME_DESCRIPTOR);
-			if (i != -1)
-				rootLocation = rootLocation.substring(0, i);
+			location = location.toAbsolutePath();
+			if (location.endsWith(Constants.FEATURE_FILENAME_DESCRIPTOR)) {
+				location = location.getParent();
+			}
+			rootLocation = location.toString() + File.separator;
```

Comment:
This is another reason why using Strings to model local file-system paths is a really bad idea, besides the forward-/backwards-slash topic: One always has to pay attention if trailing separators are present or not...

But since that method is called in a lot of places, I'll leave that for a follow-up PR.